emails/models.py

```diff
@@ -62,6 +66,15 @@ def make_relay_address(user, num_tries=0):
         if num_tries >= 5:
             raise CannotMakeAddressException
         relay_address = RelayAddress.objects.create(user=user)
+        address_contains_badword = any(
+            badword in relay_address.address
+            for badword in emails_config.badwords
+        )
+        if address_contains_badword:
+            relay_address.delete()
+            num_tries += 1
+            return RelayAddress.make_relay_address(user, num_tries)
```

Comment:
I recommend putting this check with
`if address_already_deleted > 0:`
so that we can keep this as a single recursion.